src/commands/lambda/function.ts

```diff
@@ -141,12 +162,50 @@ const calculateUpdateRequest = (
     needsUpdate = true
     updateRequest.Handler = HANDLER_LOCATION[runtime]
   }
-  const layerARNs = (config.Layers ?? []).map((layer) => layer.Arn ?? '')
-  const fullLayerARN = `${layerARN}:${settings.layerVersion}`
-  if (!layerARNs.includes(fullLayerARN)) {
+  let fullLambdaLibraryLayerARN: string | undefined
+  if (settings.layerVersion !== undefined) {
+    fullLambdaLibraryLayerARN = `${lambdaLibraryLayerArn}:${settings.layerVersion}`
+  }
+  let fullExtensionLayerARN: string | undefined
+  if (settings.extensionVersion !== undefined) {
+    fullExtensionLayerARN = `${lambdaExtensionLayerArn}:${settings.extensionVersion}`
+  }
+  if (apiKey !== undefined && env.DD_API_KEY === undefined) {
+    needsUpdate = true
+    newEnvVars.DD_API_KEY = apiKey
+  }
+  if (apiKmsKey !== undefined && env.DD_KMS_API_KEY === undefined) {
     needsUpdate = true
-    // Remove any other versions of the layer
-    updateRequest.Layers = [...layerARNs.filter((l) => !l.startsWith(layerARN)), fullLayerARN]
+    newEnvVars.DD_KMS_API_KEY = apiKmsKey
+  }
+  let layerARNs = (config.Layers ?? []).map((layer) => layer.Arn ?? '')
+  let needsLayerUpdate = false
+  if (fullLambdaLibraryLayerARN !== undefined) {
+    if (!layerARNs.includes(fullLambdaLibraryLayerARN)) {
+      needsUpdate = true
+      needsLayerUpdate = true
+      // Remove any other versions of the layer
+      layerARNs = [...layerARNs.filter((l) => !l.startsWith(lambdaLibraryLayerArn)), fullLambdaLibraryLayerARN]
+    }
+  }
+  if (fullExtensionLayerARN !== undefined) {
```

Comment:
nit: seems like most of this logic is duplicated from line 183. Could it be moved into a function?

Reply:
yeah I agree that its crowding the function, I moved it to `determineIfLayerShouldBeAdded`